### PR TITLE
Scrub PII data from the job request message in the diag log

### PR DESCRIPTION
**Pull request by juliobbv**
There are some agent environmental variables (build and release), and a property in the repository info, which may contain PII (Personal Identifiable Information) data. Scrub those places from the agent diagnostics log.

Example of how the scrub looks like in practice:

**Build:**
```
    "build.requestedForEmail": {
      "value": "***"
    }
```

**Release:**
```
    "release.requestedForEmail": {
      "value": "***"
    },
```

### CODE REVIEW

File: src/Agent.Worker/Variables.cs [hunk 1]
```diff
@@ -169,6 +169,20 @@ public Variables(IHostContext hostContext, IDictionary<string, VariableValue> co
 
         public string System_TFCollectionUrl => Get(WellKnownDistributedTaskVariables.TFCollectionUrl);
 
+        public static readonly HashSet<string> PiiVariables = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "Build.AuthorizeAs",
```
Comment (TingluoHuang) [inline src/Agent.Worker/Variables.cs]: What about all `releases.` variables?

Comment (juliobbv) [reply src/Agent.Worker/Variables.cs]: This is a very good point. So far, I've only looked at the build documentation, and I forgot about release. I'll do so as my next task.

File: src/Agent.Worker/Worker.cs [hunk 2]
```diff
@@ -102,6 +103,61 @@ public async Task<int> RunAsync(string pipeIn, string pipeOut)
...
+            }
+
+            var scrubbedRepositories = new List<Pipelines.RepositoryResource>();
+
+            // Scrub the repository resources
+            foreach (var repository in message.Resources.Repositories)
+            {
+                Pipelines.RepositoryResource scrubbedRepository = repository.Clone();
+
+                scrubbedRepository.Properties.Set("VersionInfo", "***");
```
Comment (TingluoHuang) [inline src/Agent.Worker/Worker.cs]: there is a const for `versioninfo`, `pipelines.repositorypropertynames.versioninfo`, i think.

Comment (TingluoHuang) [reply src/Agent.Worker/Worker.cs]: PS. VersionInfo is actually a class not a single string. :)

File: src/Agent.Worker/Worker.cs [hunk 3]
```diff
@@ -102,6 +103,61 @@ public async Task<int> RunAsync(string pipeIn, string pipeOut)
...
+                scrubbedRepositories.Add(scrubbedRepository);
+            }
+
+            var scrubbedJobResources = new Pipelines.JobResources();
+
+            scrubbedJobResources.Containers.AddRange(message.Resources.Containers);
```
Comment (TingluoHuang) [inline src/Agent.Worker/Worker.cs]: we don't want to do something for container?

File: src/Agent.Worker/Worker.cs [hunk 4]
```diff
@@ -102,6 +103,61 @@ public async Task<int> RunAsync(string pipeIn, string pipeOut)
...
+            scrubbedJobResources.Endpoints.AddRange(message.Resources.Endpoints);
```
Comment (TingluoHuang) [inline src/Agent.Worker/Worker.cs]: we don't want to do something for endpoint?

File: src/Agent.Worker/Worker.cs [hunk 5]
```diff
@@ -102,6 +103,61 @@ public async Task<int> RunAsync(string pipeIn, string pipeOut)
...
+            ArgUtil.NotNull(message, nameof(message));
+
+            var scrubbedVariables = new Dictionary<string, VariableValue>();
+
+            // Scrub the known PII variables
+            foreach (var variable in message.Variables)
+            {
+                if (Variables.PiiVariables.Contains(variable.Key))
+                {
+                    scrubbedVariables[variable.Key] = "***";
```
Comment (TingluoHuang) [inline src/Agent.Worker/Worker.cs]: [pii] instead of `***`

File: src/Agent.Worker/Worker.cs [hunk 6]
```diff
@@ -58,9 +59,9 @@ public async Task<int> RunAsync(string pipeIn, string pipeOut)
                 // Initialize the secret masker and set the thread culture.
                 InitializeSecretMasker(jobMessage);
                 SetCulture(jobMessage);
-
+
```
Comment (TingluoHuang) [inline src/Agent.Worker/Worker.cs]: extra space. :D

File: src/Agent.Worker/Variables.cs [hunk 7]
```diff
@@ -169,6 +169,24 @@ public Variables(IHostContext hostContext, IDictionary<string, VariableValue> co
...
+            "Build.SourceBranch",
+            "Build.SourceBranchName",
+            "Build.SourceTfvcShelveset",
+            "Build.SourceVersion",
+            "Build.SourceVersionAuthor",
+            "Job.AuthorizeAs",
+            "Release.Deployment.RequestedFor",
+            "Release.Deployment.RequestedForEmail",
+            "Release.RequestedFor",
+            "Release.RequestedForEmail",
```
Comment (TingluoHuang) [inline src/Agent.Worker/Variables.cs]: make sure add all related release variables, like `release.artifact.xxxx`
https://docs.microsoft.com/en-us/azure/devops/pipelines/release/variables?view=azure-devops&tabs=batch

File: src/Agent.Worker/Worker.cs (hunk 3 above)
Comment (juliobbv) [reply src/Agent.Worker/Worker.cs]: Ting and I talked with Matt on this one. We won't scrub the container info the within the scope of this PR.

File: src/Agent.Worker/Worker.cs (hunk 4 above)
Comment (juliobbv) [reply src/Agent.Worker/Worker.cs]: Same here. This PR will just focus on the env vars and repositories.

**Pull request merged**